src/screens/Exchange/ExchangeConfirm.js

```diff
@@ -345,12 +351,19 @@ class ExchangeConfirmScreen extends React.Component<Props, State> {
 
               </AllowanceWrapper>
             }
-            <ButtonText
+            {!hasSmartWallet && <ButtonText
               buttonText="Legacy Wallet"
               rightIconProps={{ name: 'selector', style: { fontSize: 16 } }}
               onPress={() => navigation.navigate(EXCHANGE_RECEIVE_EXPLAINED)}
               wrapperStyle={{ marginTop: 0 }}
-            />
+            />}
+            {!!hasSmartWallet &&
+              <SettingsWrapper>
+                <BaseText secondary regular center style={{ marginBottom: 0 }}>
+                  The assets will be transferred to your Smart Wallet.
```

Comment:
Shouldn't this phrase depend on the wallet you selected to receive to?

Reply:
With the latest changes the smart wallet will be the default receiver (image1). We agreed to have this button only to guide users (edge-case legacy only) to create a smart wallet account. (images 2, 3)